Tools/clinic/libclinic/clanguage.py

```diff
@@ -396,6 +396,11 @@ def render_function(
         first_optional = len(selfless)
         positional = selfless and selfless[-1].is_positional_only()
         has_option_groups = False
+        requires_defining_class = (len(selfless)
+                                   and isinstance(selfless[0].converter,
+                                                  defining_class_converter))
+        has_varag_hack = all(p.is_positional_only() or p.is_vararg()
```

Comment:
Let's not call it a hack unless it is a temporary change :)

Reply:
Suggestions welcome:)  But I think that *might* be a temporary change, as in future we can extend this fix to positional-or-keyword arguments.

Reply:
Braindump:

- `fast_vararg`
- `pass_vararg_directly`
- `avoid_temp_tuple`

Other suggestions are welcome! I don't like any of mine 😆